src/cascadia/LocalTests_SettingsModel/DeserializationTests.cpp

```diff
@@ -2023,6 +2030,189 @@ namespace SettingsModelLocalTests
+        const auto& nested{ actionsByName.TryLookup(L"nested command") };
+        VERIFY_IS_NOT_NULL(nested);
+        VERIFY_IS_TRUE(nested.HasNestedCommands());
+    }
+
+    void DeserializationTests::FragmentActionNestedNoName()
+    {
+        // Basically the same as TestNestedCommandWithoutName
+        static constexpr std::wstring_view fragmentSource{ L"fragment" };
+        static constexpr std::string_view fragmentJson{ R"({
+            "actions": [
+                {
+                    "commands": [
+                        {
+                            "name": "child1",
+                            "command": { "action": "newTab", "commandline": "ssh me@first.com" }
+                        },
+                        {
+                            "name": "child2",
+                            "command": { "action": "newTab", "commandline": "ssh me@second.com" }
+                        }
+                    ]
+                },
+            ]
+        })" };
+
+        implementation::SettingsLoader loader{ std::string_view{}, DefaultJson };
+        loader.MergeInboxIntoUserSettings();
+        loader.MergeFragmentIntoUserSettings(winrt::hstring{ fragmentSource }, fragmentJson);
+        loader.FinalizeLayering();
+
+        const auto settings = winrt::make_self<implementation::CascadiaSettings>(std::move(loader));
+        VERIFY_ARE_EQUAL(0u, settings->Warnings().Size());
+    }
+    void DeserializationTests::FragmentActionIterable()
+    {
+        static constexpr std::wstring_view fragmentSource{ L"fragment" };
+        static constexpr std::string_view fragmentJson{ R"({
+            "actions": [
+                {
+                    "name": "nested",
+                    "commands": [
+                        {
+                            "iterateOn": "schemes",
+                            "name": "${scheme.name}",
+                            "command": { "action": "setColorScheme", "colorScheme": "${scheme.name}" }
+                        }
+                    ]
+                },
+            ]
+        })" };
+
+        implementation::SettingsLoader loader{ std::string_view{}, DefaultJson };
+        loader.MergeInboxIntoUserSettings();
+        loader.MergeFragmentIntoUserSettings(winrt::hstring{ fragmentSource }, fragmentJson);
+        loader.FinalizeLayering();
+
+        const auto settings = winrt::make_self<implementation::CascadiaSettings>(std::move(loader));
+
+        const auto actionMap = winrt::get_self<implementation::ActionMap>(settings->GlobalSettings().ActionMap());
+        const auto actionsByName = actionMap->NameMap();
+        const auto& nested{ actionsByName.TryLookup(L"nested") };
+        VERIFY_IS_NOT_NULL(nested);
+        VERIFY_IS_TRUE(nested.HasNestedCommands());
+        VERIFY_ARE_EQUAL(settings->GlobalSettings().ColorSchemes().Size(), nested.NestedCommands().Size());
+    }
+    void DeserializationTests::FragmentActionRoundtrip()

```

Comment:
i like the round-trip test verifying we didn't serialize someone else's action. clever.